dockstore-webservice/src/main/java/io/dockstore/webservice/core/Organization.java

```diff
@@ -309,7 +309,7 @@ public void setDisplayName(String displayName) {
         this.displayName = displayName;
     }
 
-    public enum ApplicationState { PENDING, REJECTED, APPROVED
+    public enum ApplicationState { PENDING, REJECTED, APPROVED, SPECIAL
```

Comment:
Could also just subclass Organization for your "special" Organization that holds categories

Reply:
If we used a subclass for special orgs (via a JPA single table inheritance scheme with dtype column?), to hide them, we'd either have to set their application state to PENDING or REJECTED or change the org visibility logic.  The SPECIAL application state achieves the hiding with no code changes (beyond the additional enum value), and it kinda makes sense, since special orgs don't go through an application process.  I probably favor the SPECIAL enum, but not strongly: it was super-easy and very safe to implement. 

Regarding how special orgs are created, the SPECIAL enum solution might be less complicated, too.  If an Organization with SPECIAL application status is POSTed to the /organizations endpoint, it gets created with SPECIAL status.

Reply:
> via a JPA single table inheritance scheme with dtype column 

Yeah

I don't feel strongly about this. But I do feel that it is cleaner to make JPA aware that these organizations are special. You'd be able to do queries like https://github.com/dockstore/dockstore/pull/4456/files#diff-d8f14acba6dc43ef91cb6e8f2c3bc50a68b3627f24d5d16c7f40d9d02c301c98R66 using type JPQL queries https://stackoverflow.com/questions/21737435/subclass-discrimination-type-using-a-class-member-in-jpa-jpql

Might also be able to avoid the conversion in https://github.com/dockstore/dockstore/pull/4456/files#diff-4a1e697b2cba80f86858316c76424c8029f0b3f82dff1b3906db8f8c41576324R448 if JPA understands that these special organisations have categories instead of collections in a type-safe manner.

Reply:
Another approach is that each Organization gets some new properties/columns/accessors to control visibility and to trigger category creation.  Let's call them getHidden() and getIsCategorizer().  You could even get crazy and have getVisibility() with values of DEFAULT, VISIBLE, and HIDDEN.  This approach allows finer control, and organizations can wear multiple hats (when more hats exist someday).